spec/models/unpublishing_alert_spec.rb

```diff
@@ -0,0 +1,261 @@
+  let(:subscriber_list_response) do
+    {
+      "subscriber_list" => {
+        "id" => subscriber_list_uuid,
+        "title" => "Subscriber List Title",
+      },
+    }
+  end
+  let(:alert_api) { double(:alert_api, bulk_unsubscribe: nil, find_subscriber_list: subscriber_list_response) }
+
+  before :each do
+    allow(LockHandler).to receive(:new).and_return(fake_lock_handler)
+    allow(Services).to receive(:email_api_client).and_return(alert_api)
+  end
+
+  describe "#trigger" do
+    describe "by published_in_error_with_url event" do
+      let(:unpublishing_scenario) { :published_in_error_with_url }
+      let(:alternative_path) { "/foo/alternative" }
+      let(:document) { published_in_error_payload }
+      let(:email_markdown) do
+        <<~UNPUBLISHING_MESSAGE
+          Change made:
+          This page was removed from GOV.UK because it was published in error. It’s been replaced by: https://www.test.gov.uk/foo/alternative
+
+          Time updated:
+          #{formatted_time}
+        UNPUBLISHING_MESSAGE
+      end
+      it "logs an unsubscription notification" do
+        unpublishing_alert.trigger
+
+        expect(logger).to have_received(:info).with(
+          "Received unsubscription notification for #{document['title']}, unpublishing_scenario: #{unpublishing_scenario}, full payload: #{document}",
+        )
+      end
+
+      it "sends a bulk unsubscribe request to the Email Alert API" do
+        unpublishing_alert.trigger
+
+        expect(alert_api).to have_received(:bulk_unsubscribe).with(
+          {
+            "subscriber_list_id" => subscriber_list_uuid,
+            "body" => email_markdown.strip,
+            "sender_message_id" => sender_message_id,
+          }.to_json,
+        )
+      end
+
+      it "logs if it recieves a conflict" do
+        allow(alert_api).to receive(:bulk_unsubscribe).and_raise(GdsApi::HTTPConflict.new(409))
+        unpublishing_alert.trigger
+
+        expect(logger).to have_received(:info).with(
+          "email-alert-api returned conflict for #{document['content_id']}, #{document['base_path']}, #{document['public_updated_at']}",
+        )
+      end
+
+      it "logs if it recieves an unprocessable entity" do
+        allow(alert_api).to receive(:bulk_unsubscribe).and_raise(GdsApi::HTTPUnprocessableEntity.new(422))
+        unpublishing_alert.trigger
+
+        expect(logger).to have_received(:info).with(
+          "email-alert-api returned unprocessable entity for #{document['content_id']}, #{document['base_path']}, #{document['public_updated_at']}",
+        )
+      end
+
+      it "logs if it recieves not found" do
+        allow(alert_api).to receive(:bulk_unsubscribe).and_raise(GdsApi::HTTPNotFound.new(404))
+        unpublishing_alert.trigger
+
+        expect(logger).to have_received(:info).with(
+          "email-alert-api returned not_found for #{document['content_id']}, #{document['base_path']}, #{document['public_updated_at']}",
+        )
+      end
```

Comment:
This was also going to be a shared example, right?

Reply:
Yeah was just looking at those, can do those too if we want. 1 sec